cpp/cmake/thirdparty/get_raft.cmake

```diff
@@ -0,0 +1,50 @@
+#=============================================================================
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_raft)
+
+    set(oneValueArgs VERSION FORK PINNED_TAG)
+    cmake_parse_arguments(PKG "${options}" "${oneValueArgs}"
+                          "${multiValueArgs}" ${ARGN} )
+
+    rapids_cpm_find(raft ${PKG_VERSION}
+      GLOBAL_TARGETS      raft::raft
+      BUILD_EXPORT_SET    cuml-exports
+      INSTALL_EXPORT_SET  cuml-exports
+        CPM_ARGS
+            GIT_REPOSITORY https://github.com/${PKG_FORK}/raft.git
+            GIT_TAG        ${PKG_PINNED_TAG}
+            SOURCE_SUBDIR  cpp
+            OPTIONS
+              "BUILD_TESTS OFF"
+
+    )
+
+    message(VERBOSE "CUML: Using RAFT located in ${raft_SOURCE_DIR}")
+
+endfunction()
+
+set(CUML_MIN_VERSION_raft "${CUML_VERSION_MAJOR}.${CUML_VERSION_MINOR}")
+
+# Change pinned tag here to test a commit in CI
+# To use a different RAFT locally, set the CMake variable
+# RPM_raft_SOURCE=/path/to/local/raft
+find_and_configure_raft(VERSION    ${CUML_MIN_VERSION_raft}
+                        FORK       rapidsai
+                        PINNED_TAG fe5f40aad9159c60fbc53f80da689999c1e86203
```

Comment:
Is this the same commit as [cugraph is using](https://github.com/rapidsai/cugraph/pull/1585/files/91d1ea220c6998c2d7839f9b7ddf92f55b5c5036..693074d354a41ed0cd076b73806faece05a1a49c#diff-7bbc4aa5eba7917d781a8f8304caae99c71d892568ca359e5391a75a401cc8ebR44)? Can they be pinned to the same release branch now?

Reply:
Yeah I think pinning them to the main development branch is fine now, will just raise awareness with the teams. Right now just added the commit to be sure CI was testing the correct thing (and then got style issues :unamused: )

Reply:
I was chatting with the team about just pinning to `branch-21.06` directly, and so far response has been positive so I think it is the way to go, will be pushing the change in a bit once the branch naming migration is done to test CI.